grasa_event_locator/views.py

```diff
@@ -254,6 +263,17 @@ def provider(request):
                         u.org_name = request.POST['changename']
                         u.save()
                         rebuildIndex.rebuildWhooshIndex()
+                if request.POST.get('current') and (request.POST.get('new') == request.POST.get('confirm')):
+                    current = request.POST['current']
+                    new = request.POST['new']
+                    if request.user.check_password(current):
+                        request.user.set_password(new)
+                        request.user.save()
```

Comment:
Could you walk me through what is happening in these three lines?

Reply:
The same thing happens in the admin page, see the comments in there. Basically though, check_password checks whether the inputted current password (current) is actually the user's password. If it exists (in other words, if it comes out to be true), then set the password to the new password, then save the user (which is required for any of these type of changes to apply).

Reply:
And to clarify, in any instance of request.POST.get('something') or request.POST['something'], 'something', just refers to the "name='something'", which is a part of the input tag in the html for the corresponding field.